ipatests/test_integration/test_acme.py

```diff
@@ -769,29 +824,213 @@ def test_pruning_negative_options(self):
+
+    def test_prune_cert_retention_unit(self, issue_and_expire_acme_cert):
+        """Test to prune expired certificate with retention unit option"""
+        if (tasks.get_pki_version(self.master)
+           < tasks.parse_version('11.3.0')):
+            raise pytest.skip("Certificate pruning is not available")
+        issue_and_expire_acme_cert(
+            self.master, self.clients[0], self.acme_server)
+
+        # check that the certificate issued for the client
+        result = self.master.run_command(['ipa', 'cert-find'])
+        assert f'CN={self.clients[0].hostname}' in result.stdout_text
+
+        # enable pruning
+        self.master.run_command(['ipa-acme-manage', 'pruning', '--enable'])
+
+        # certretention set to 5 min
+        self.master.run_command(
+            ['ipa-acme-manage', 'pruning',
+             '--certretention=5', '--certretentionunit=minute']
+        )
+        self.master.run_command(['ipactl', 'restart'])
+        # self.master.run_command(
+        #     ['ipa-acme-manage', 'pruning', '--run']
+        # )
+
+        # check after 2 minutes that cert is still present
+        time.sleep(120)
+        result = self.master.run_command(['ipa', 'cert-find'])
+        assert f'CN={self.clients[0].hostname}' in result.stdout_text
+
+        # sleep for more than 5 min and check if expired cert is removed
+        time.sleep(400)
+        result = self.master.run_command(['ipa', 'cert-find'])
+        assert f'CN={self.clients[0].hostname}' not in result.stdout_text
+
+    def test_prune_cert_search_size_limit(self, issue_and_expire_acme_cert):
+        """Test to prune expired certificate with search size limit option"""
+        if (tasks.get_pki_version(self.master)
+           < tasks.parse_version('11.3.0')):
+            raise pytest.skip("Certificate pruning is not available")
+        no_of_cert = 10
+        search_size_limit = 5
+        issue_and_expire_acme_cert(
+            self.master, self.clients[0], self.acme_server, no_of_cert)
+
+        # check that the certificate issued for the client
+        result = self.master.run_command(['ipa', 'cert-find'])
+        assert f'CN={self.clients[0].hostname}' in result.stdout_text
+        assert f'Number of entries returned {10 + no_of_cert}'
+
+        # enable pruning
+        self.master.run_command(['ipa-acme-manage', 'pruning', '--enable'])
+
+        # certretention set to 5 min
+        self.master.run_command(
+            ['ipa-acme-manage', 'pruning',
+             f'--certsearchsizelimit={search_size_limit}',
+             '--certsearchtimelimit=100']
+        )
+        self.master.run_command(['ipactl', 'restart'])
+        time.sleep(100)
+
+        # prune the certificates
+        self.master.run_command(['ipa-acme-manage', 'pruning', '--run'])
+
+        # check if 5 expired cert is removed
+        result = self.master.run_command(['ipa', 'cert-find'])
+        assert f'Number of entries returned {20 - search_size_limit}'
```

Comment:
I think you want:  {no_of_cert - search_size_limit}

Reply:
10 we are issuing acme certs + 9 subsystemcert + master cert = 20